src/main/main.ts

```diff
@@ -192,8 +192,11 @@ app.on('ready', () => {
 app.on("web-contents-created", (_event: any, webContents: WebContents) => {
     // Prevent navigation
     webContents.on('will-navigate', (event: Event, navigationUrl) => {
-        console.warn(`Navigation is not allowed; attempted navigation to: ${navigationUrl}`);
-        event.preventDefault();
+        const jlabBaseUrl = `http://localhost:${appConfig.jlabPort}/`;
+        if (!navigationUrl.startsWith(jlabBaseUrl)) {
+            console.warn(`Navigation is not allowed; attempted navigation to: ${navigationUrl}`);
+            event.preventDefault();
+        }
```

Comment:
Need to test if this does not re-introduce https://github.com/jupyterlab/jupyterlab-desktop/issues/332. If it does we can try `nativeWindowOpen` instead.

Reply:
updated, and now allowing links to external sites and different URLs on the running JupyterLab server since they open in a separate window.